ts/reviewer/answering.ts

```diff
@@ -53,11 +59,18 @@ function packCustomData(
 
 export async function mutateNextCardStates(
     key: string,
-    mutator: (states: Scheduler.SchedulingStates, customData: CustomDataStates) => void,
+    mutator: (
+        states: Scheduler.SchedulingStates,
+        customData: CustomDataStates,
+        ctx: Scheduler.SchedulingContext,
+    ) => void,
 ): Promise<void> {
-    const states = await getSchedulingStates();
+    const [states, ctx] = await Promise.all([
+        getSchedulingStates(),
+        getSchedulingContext(),
+    ]);
```

Comment:
I'd like to try keep the number of separate requests to a minimum, as there's potentially a large round trip time (eg, if AnkiWeb adds support for this, there can be a RTT of 300-400ms in some parts of the world). Doing the requests in parallel does significantly improve the situation, but it's still an extra request that needs to be processed.

How about packing these two values into a container struct and returning that in a single method? We could also potentially return the context in get_queued_cards(), like we already do for the states.

Reply:
That's interesting. I've never thought about how it'd work on AnkiWeb.